describeconfigs.go

```
@@ -0,0 +1,157 @@
+
+	for i, t := range req.Resources {
+		configNames := make([]string, len(t.ConfigNames))
+		for _, v := range t.ConfigNames {
+			configNames = append(configNames, v)
+		}
+		resources[i] = describeconfigs.RequestResource{
+			ResourceType: t.ResourceType,
+			ResourceName: t.ResourceName,
+			ConfigNames:  configNames,
+		}
+	}
+
+	m, err := c.roundTrip(ctx, req.Addr, &describeconfigs.Request{
+		Resources:            resources,
+		IncludeSynonyms:      req.IncludeSynonyms,
+		IncludeDocumentation: req.IncludeDocumentation,
+	})
+
+	if err != nil {
+		return nil, fmt.Errorf("kafka.(*Client).DescribeConfigs: %w", err)
+	}
+
+	res := m.(*describeconfigs.Response)
+	ret := &DescribeConfigsResponse{
+		Throttle:  makeDuration(res.ThrottleTimeMs),
+		Resources: make([]DescribeConfigResponseResource, len(res.Resources)),
+	}
+
+	for _, t := range res.Resources {
+
+		configEntries := make([]DescribeConfigResponseConfigEntry, len(t.ConfigEntries))
+		for _, v := range t.ConfigEntries {
+
+			configSynonyms := make([]DescribeConfigResponseConfigSynonym, len(v.ConfigSynonyms))
+			for _, cs := range v.ConfigSynonyms {
+				configSynonyms = append(configSynonyms, DescribeConfigResponseConfigSynonym{
+					ConfigName:   cs.ConfigName,
+					ConfigValue:  cs.ConfigValue,
+					ConfigSource: cs.ConfigSource,
+				})
+			}
+
+			configEntries = append(configEntries, DescribeConfigResponseConfigEntry{
+				ConfigName:          v.ConfigName,
+				ConfigValue:         v.ConfigValue,
+				ReadOnly:            v.ReadOnly,
+				IsDefault:           v.IsDefault,
+				IsSensitive:         v.IsSensitive,
+				ConfigSynonyms:      configSynonyms,
+				ConfigType:          v.ConfigType,
+				ConfigDocumentation: v.ConfigDocumentation,
+			})
+		}
+
+		resource := DescribeConfigResponseResource{
+			ResourceType:  t.ResourceType,
+			ResourceName:  t.ResourceName,
+			ErrorCode:     t.ErrorCode,
+			ErrorMessage:  t.ErrorMessage,
```

can we return an error here using `makeError`